server/sonar-server/src/test/java/org/sonar/server/computation/issue/RuleRepositoryImplTest.java

```diff
@@ -19,37 +19,223 @@
  */
 package org.sonar.server.computation.issue;
 
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.rule.RuleStatus;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.rule.RuleDao;
+import org.sonar.db.rule.RuleDto;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.guava.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyBoolean;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
 import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 import static org.mockito.internal.verification.VerificationModeFactory.times;
-import static org.sonar.db.rule.RuleTesting.XOO_X1;
-import static org.sonar.db.rule.RuleTesting.XOO_X2;
 
 public class RuleRepositoryImplTest {
 
-  RuleCacheLoader cacheLoader = mock(RuleCacheLoader.class);
-  RuleRepositoryImpl underTest = new RuleRepositoryImpl(cacheLoader);
+  private static final RuleDto AB_RULE = createABRuleDto();
+  private static final RuleKey AC_RULE_KEY = RuleKey.of("a", "c");
+  private static final int AC_RULE_ID = 684;
+
+  @org.junit.Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  private DbClient dbClient = mock(DbClient.class);
+  private DbSession dbSession = mock(DbSession.class);
+  private RuleDao ruleDao = mock(RuleDao.class);
```

Comment:
I imagine it's because you wanted to test that the call to the db is only do once.
To do that with dbTester, you can for instance to a first call to load all rules from db, then remove them from the db and check that you can still get some rules.